### ISSUE TITLE
[java] ClassWithOnlyPrivateConstructorsShouldBeFinal false positive with Lombok's `@NoArgsConstructor`

### ISSUE THREAD

**Issue by Belle-PL** [pmd/pmd#4188]
**Affects PMD Version:6.51.0** 

**Rule:ClassWithOnlyPrivateConstructorsShouldBeFinal**

Please provide the rule name and a link to the rule documentation:
<https://pmd.sourceforge.io/pmd-6.51.0/pmd_rules_java_design.html#classwithonlyprivateconstructorsshouldbefinal>

**Description:**
Hi, I found a false positive about this rule, please read the following code example. PMD should not report a warning for this class because `@NoArgsConstructor` can add a constructor for this class. Hence, I think this is a false positive. 

* https://projectlombok.org/features/constructor
* https://projectlombok.org/api/lombok/NoArgsConstructor

**Code Sample demonstrating the issue:**

```java
import lombok.NoArgsConstructor;
@NoArgsConstructor
public class C {
    private C(String[] arr, long in, String res) {}
}
```

**Expected outcome:**

PMD reports a violation at line 3, but that's wrong. That's a false positive.

**Running PMD through:** *[Maven]*
